DOPS-101 Add trussworks bootstrap to devops example

akagr opened the pull request:
# Proposed Changes

After research, I've found https://github.com/trussworks/terraform-aws-bootstrap to be the easiest solution for bootstrapping Terraform projects. Using this module, we can add the bootstrap infrastructure to the repo and manage it going forward.

## How it works

The bootstrap module has a local backend. We need to first apply the bootstrap module to generate the resources like S3 bucket, DynamoDB table etc.

These resources can then be used in the backend of main Terraform project.

✅ Add bootstrap project
✅ Updated `README.md` with new setup instructions for Terraform project
✅ Added the bootstrap state to the repo. This makes sure the terraform project is good to go without bootstrapping (if using default aliases and prefix)

akagr committed:
DOPS-101 Add bootstrap module for terraform project

akagr committed:
DOPS-101 Add bootstrap and backend details to terrform project

Updated README to point to bootstrap project for setting up terraform
backend.

Updated main terraform project's backend with the S3 and DynamoDB
tables generated using default bootstrap values.

modus-jenkins commented:
Atlantis commands can't be run on fork pull requests. To enable, set --allow-fork-prs  or, to disable this message, set --silence-fork-pr-errors

akagr commented:
Noticed the failing pipeline. Working on that.

akagr committed:
DOPS-101 Add variable for account alias for main terraform project

This alias needs to match the one used in bootstrap project to use
correct names for S3 bucket and DynamoDB table.

modus-jenkins commented:
Atlantis commands can't be run on fork pull requests. To enable, set --allow-fork-prs  or, to disable this message, set --silence-fork-pr-errors

akagr committed:
DOPS-101 Bootstrap terraform state for sandbox account

We validate the backend in our CI and need the S3 and DynamoDB resources
to exist for that to pass.

modus-jenkins commented:
Atlantis commands can't be run on fork pull requests. To enable, set --allow-fork-prs  or, to disable this message, set --silence-fork-pr-errors

akagr commented:
Closing this pull request because atlantis doesn't work for requests from forks. Create a new PR https://github.com/ModusCreateOrg/devops-infra-demo/pull/64 from a branch in this repo instead.
